driver-core/src/main/java/com/datastax/driver/core/EventDebouncer.java

```diff
@@ -166,7 +169,7 @@ private void scheduleImmediateDelivery() {
         while (state == State.RUNNING) {
             DeliveryAttempt previous = immediateDelivery.get();
             if (previous != null && !previous.isDone())
-                return;
+                previous.cancel();
```

Comment:
deliverEvents does not handle interruptions, so this is essentially a no-op if previous is already executing.